### PR TITLE
Objectives support

### LINKED ISSUE #3: Scoreboard objectives support

### PR DESCRIPTION
**Pull request by vytskalt**
TODO:
- [x] finalize api
- [x] add javadocs
- [x] no-op
- [x] packet adapter stuff
- [x] api implementation
- [x] add examples to readme
- [x] make sidebar reuse the packet adapter internally

closes #3

### CODE REVIEW

**Comment by huanmeng-qwq** [comment]
Looking forward to achieving it as soon as possible

**Comment by huanmeng-qwq** [comment]
Can you promote the completion of this feature? @vytskalt

**Comment by vytskalt** [comment]
> Can you promote the completion of this feature? @vytskalt

I don't have much time due to school atm unfortunately, but I'll try to complete this within a week

File: versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java [hunk 1]
```diff
@@ -87,6 +91,11 @@ public void removeSidebar(@NotNull Iterable<Player> players) {
     sendPacket(players, removePacket);
   }
 
+  @Override
+  public @NotNull ObjectivesPacketAdapter<?, ?> createObjectiveAdapter(@NotNull String objectiveName) {
+    return isNativeAdventure ? new PaperObjectivePacketAdapter(this, objectiveName) : new SpigotObjectivePacketAdapter(this, objectiveName);
```
Comment (huanmeng-qwq) [inline versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: Using isNativeAdventure to determine if there may be some errors, as not only paperspot has Adventure libraries

Comment (vytskalt) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: It's for making some optimisations on paper based servers

Comment (huanmeng-qwq) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: But this cannot guarantee the existence of the class: `io.papermc.paper.adventure.AdventureComponent`, which may result in a ClassNotFoundException

Comment (vytskalt) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: isNativeAdventure specifies whether the server is paper based, so if it is true then its safe to assume AdventureComponent exists

Comment (huanmeng-qwq) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: I used a `custom Spigot`, which is not a branch of paper, but it also integrates the `Adventure` library and does not have the `AdventureComponent` class

Comment (vytskalt) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: isNativeAdventure specifically checks for paper related classes so not an issue

Comment (huanmeng-qwq) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: Oh, I'm very sorry, I didn't see the judgment on the PaperAdventure class above. I'm very sorry, I'm really sorry, it was caused by my carelessness

Comment (vytskalt) [reply versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java:96]: its fine lol

**Comment by huanmeng-qwq** [comment]
Is it ready to use?

**Comment by huanmeng-qwq** [comment]
Looking forward to successfully running on 1.8.8!

**Comment by vytskalt** [comment]
> Looking forward to successfully running on 1.8.8!

I've implemented it for 1.8.8, seems to work fine from my limited testing. You can change the version of the dependency to `087d086945` to get the changes. Example is [here](https://github.com/MegavexNetwork/scoreboard-library/tree/feat/objectives#objectivemanager). Can you test it out if it works for you?

**Comment by huanmeng-qwq** [comment]
It seems that the version of packetevent I am using has not been implemented yet?
https://github.com/MegavexNetwork/scoreboard-library/blob/087d086945df2535ee843c1f5cb7fe61ccd24923/versions/packetevents/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/packetevents/PacketAdapterImpl.java#L55-L58

**Comment by huanmeng-qwq** [comment]
At present, there seems to be no way to provide a specific player display method similar to ScoreboardTeam#display(Player, TeamDisplay)

**Comment by vytskalt** [comment]
Yeah I havent implemented it for packetevents yet, but you can use the native 1.8.8 packet adapter

**Comment by huanmeng-qwq** [comment]
findAndLoadImplementationClass cannot correctly obtain the version by using 'Bukkit.getServer().getBukkitVersion()'. I am using a custom Spigot, can you try switching to 'Bukkit.getServer().getClass().getPackage()'?

**Comment by vytskalt** [comment]
> findAndLoadImplementationClass cannot correctly obtain the version by using 'Bukkit.getServer().getBukkitVersion()'. I am using a custom Spigot, can you try switching to 'Bukkit.getServer().getClass().getPackage()'?

PaperMC has recently advised to switch from using `Bukkit.getServer().getClass().getPackage()` to `Bukkit.getServer().getBukkitVersion()` as they plan to remove the version from the package
https://discord.com/channels/289587909051416579/1077385604012179486/1155459001090318346 (PaperMC discord)

What kind of values does your custom spigot return?

**Comment by huanmeng-qwq** [comment]
It's the commitId of git
example: ForkSpigot-980636d

**Comment by vytskalt** [comment]
> It's the commitId of git example: ForkSpigot-980636d

Not sure what I can do to fix it. The git hash should be in `Bukkit.getServer().getVersion()` and not `Bukkit.getVersion().getBukkitVersion()`

**Comment by huanmeng-qwq** [comment]
Okay, I will try to make it work.

**Comment by huanmeng-qwq** [comment]
```
[22:49:14 ERROR]: [address|username] <-> DownstreamBridge <-> [server] - encountered exception
java.lang.IllegalArgumentException: Objective c6ad7e1834 already exists in this scoreboard
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:220) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at net.md_5.bungee.api.score.Scoreboard.addObjective(Scoreboard.java:55) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at net.md_5.bungee.connection.DownstreamBridge.handle(DownstreamBridge.java:188) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at net.md_5.bungee.protocol.packet.ScoreboardObjective.handle(ScoreboardObjective.java:67) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at net.md_5.bungee.netty.HandlerBoss.channelRead(HandlerBoss.java:114) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

```
```
objective = objectiveManager.create(randomKey());
objective.display(slot, objective);
objective.score(nickName(), 10);
objectiveManager.addPlayer(viewer.getPlayer());

**Comment by vytskalt** [comment]
Interesting, ill check it out

**Comment by vytskalt** [comment]
> ```
> [22:49:14 ERROR]: [address|username] <-> DownstreamBridge <-> [server] - encountered exception
> java.lang.IllegalArgumentException: Objective c6ad7e1834 already exists in this scoreboard
>         at com.google.common.base.Preconditions.checkArgument(Preconditions.java:220) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at net.md_5.bungee.api.score.Scoreboard.addObjective(Scoreboard.java:55) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at net.md_5.bungee.connection.DownstreamBridge.handle(DownstreamBridge.java:188) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at net.md_5.bungee.protocol.packet.ScoreboardObjective.handle(ScoreboardObjective.java:67) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at net.md_5.bungee.netty.HandlerBoss.channelRead(HandlerBoss.java:114) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:336) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:308) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.19-R0.1-SNAPSHOT:c6ccee1:unknown]
>         at java.lang.Thread.run(Thread.java:833) ~[?:?]
> ```
> 
> ```java
> objective = objectiveManager.create(randomKey());
> objectiveManager.display(ObjectiveDisplaySlot.playerList(), objective);
> objective.score(nickName(), 10);
> objectiveManager.addPlayer(viewer);
> ```
> 
> I haven't created an objective before, so I can confirm that it's the first one

fixed

**Comment by vytskalt** [comment]
new version is `b747d4bfeb`, only thing left is to test everything and then ill merge this

**Comment by huanmeng-qwq** [comment]
image
```java
objective = objectiveManager.create(randomKey());
objectiveManager.display(ObjectiveDisplaySlot.playerList(), objective);
objective.score("wq", 20);
objectiveManager.addPlayer(player("wq"));
objectiveManager.addPlayer(player("vq"));
```
It did work, but some of it didn't meet my expectations

I only set the score for 'wq', I don't want numbers to appear after 'vq'

**Comment by vytskalt** [comment]
That might be a limitation in the vanilla client itself

**Comment by huanmeng-qwq** [comment]
Okay, so far I don't have any problems. I have tested PlayerList and BelowName

**Comment by vytskalt** [comment]
Everything appears to be working from my testing, will merge this and make a release

**Pull request merged**